src/apps/wall/scanner/ndpi_opt.dasl

```diff
@@ -0,0 +1,103 @@
+-- This module creates a wrapper around nDPI's packet processing
+-- functions in order to help LuaJIT do its optimizations
+--
+-- Specifically, it avoids a struct value return by creating wrappers
+-- that pass a struct pointer in instead. LuaJIT can handle that
+-- better than a struct return (which is not supported).
+
+module(..., package.seeall)
+
+local dasm = require("dasm")
+local ffi  = require("ffi")
+local ndpi = require("ndpi.c").lib
+
+local debug = false
+
+|.arch x64
+|.actionlist actions
+
+-- the definitions here (anchor, assemble, gen) are borrowed from lwaftr
+-- (see multi_copy.lua)
+__anchor = {}
+
+local function assemble (name, prototype, generator)
+   local Dst = dasm.new(actions)
+   generator(Dst)
+   local mcode, size = Dst:build()
+   table.insert(__anchor, mcode)
+   if debug then
+      print("mcode dump: "..name)
+      dasm.dump(mcode, size)
+   end
+   return ffi.cast(prototype, mcode)
+end
+
+local function gen_nddpw(orig_f)
```

Comment:
Possible typo: I wonder what nddp stands for, shouldn't be ndpp (ndpi_detection_process_packet)?